modules/rostests/apitests/shell32/ItemIDList.cpp

```diff
@@ -0,0 +1,149 @@
+    ok_long(hr, S_OK);
+    hr = (psf2 ? psf2->GetAttributesOf(1, &pidl2Last, &attrs2) : E_UNEXPECTED);
+    ok_long(hr, S_OK);
+
+    // There is the difference in attributes because SHSimpleIDListFromPath
+    // cannot create PIDLs to folders, only files and drives:
+    ok_long((attrs1 & SFGAO_FOLDER), 0);
+    ok_long((attrs2 & SFGAO_FOLDER), SFGAO_FOLDER);
+
+
+    // Make sure the internal details match Windows NT5+
+    LPITEMIDLIST item;
+    GetSystemDirectoryW(szPath, _countof(szPath));
+    CComHeapPtr<ITEMIDLIST> pidlSys32(SHSimpleIDListFromPath(szPath));
+    if (szPath[1] != ':' || PathFindFileNameW(szPath) <= szPath)
+    {
+        skip("Not a local directory %ls\n", szPath);
+    }
+    else if (!(LPITEMIDLIST)pidlSys32)
+    {
+        skip("?\n");
+    }
+    else
+    {
+        item = ILFindLastID(pidlSys32);
+        ok_long(item->mkid.abID[0] & 0x73, 0x30 | FILEBIT); // This is actually a file PIDL
+        ok_long(FileStruct_Att(item), 0); // Simple PIDL without attributes
+        ok_int(*(UINT*)(&item->mkid.abID[2]), 0); // No size
+
+        ILRemoveLastID(pidlSys32); // Now we should have "c:\Windows"
+        item = ILFindLastID(pidlSys32);
+        ok_long(item->mkid.abID[0] & 0x73, 0x30 | DIRBIT);
+        ok_int(*(UINT*)(&item->mkid.abID[2]), 0); // No size
+    }
+
+    WCHAR drive[4] = { szPath[0], szPath[1], L'\\', L'\0' };
+    CComHeapPtr<ITEMIDLIST> pidlDrive(SHSimpleIDListFromPath(drive));
+    if (drive[1] != ':')
+    {
+        skip("Not a local drive %ls\n", drive);
+    }
+    else if (!(LPITEMIDLIST)pidlDrive)
+    {
+        skip("?\n");
+    }
+    else
+    {
+        item = ILFindLastID(pidlDrive);
+        ok_long(item->mkid.abID[0] & 0x70, 0x20); // Something in My Computer
+        ok_char(item->mkid.abID[1] | 32, drive[0] | 32);
+    }
+
+    CComHeapPtr<ITEMIDLIST> pidlVirt(SHSimpleIDListFromPath(L"x:\\IDontExist"));
+    if (!(LPITEMIDLIST)pidlVirt)
+    {
+        skip("?\n");
+    }
+    else
+    {
+        item = ILFindLastID(pidlVirt);
+        ok_long(item->mkid.abID[0] & 0x73, 0x30 | FILEBIT); // Yes, a file
+        ok_long(FileStruct_Att(item), 0); // Simple PIDL, no attributes
+        ok_int(*(UINT*)(&item->mkid.abID[2]), 0); // No size
+
+        ILRemoveLastID(pidlVirt); // "x:\"
+        item = ILFindLastID(pidlVirt);
+        ok_long(item->mkid.abID[0] & 0x70, 0x20); // Something in My Computer
+        ok_char(item->mkid.abID[1] | 32, 'x' | 32); // x:
+    }
+}
+
+START_TEST(ILCreateFromPath)
+{
+    WCHAR szPath[MAX_PATH];
+    LPITEMIDLIST item;
+
+    ok_ptr(ILCreateFromPathW(L"c:\\IDontExist"), NULL);
+
+    GetSystemDirectoryW(szPath, _countof(szPath));
+    CComHeapPtr<ITEMIDLIST> pidlDir(ILCreateFromPathW(szPath));
+    if (szPath[1] != ':' || PathFindFileNameW(szPath) <= szPath)
+    {
+        skip("Not a local directory %ls\n", szPath);
+    }
+    else if (!(LPITEMIDLIST)pidlDir)
+    {
+        skip("?\n");
```

Comment:
You may try to write some more meaningful message here, something like "Failed to create ITEMIDLIST"
(same comment for all those other "?\n" messages)

Reply:
If you see the existing code above, it does not even try to check if the pidl is created.

I don't expect this to actually fail. Not having the memory to allocate 200ish bytes is unlikely and system32 not being a valid path is never going to happen.